Based on Fedora patch 0002-redis-2.8.18-deps-library-fPIC-performance-tuning.patch, rebased for 3.2.0.

diff -Naurp a/deps/geohash-int/Makefile b/deps/geohash-int/Makefile
--- a/deps/geohash-int/Makefile	2016-05-06 02:11:36.000000000 -0500
+++ b/deps/geohash-int/Makefile	2016-06-01 18:06:58.691789064 -0500
@@ -2,7 +2,7 @@ STD=
 WARN= -Wall
 OPT= -O2
 
-R_CFLAGS= $(STD) $(WARN) $(OPT) $(DEBUG) $(CFLAGS)
+R_CFLAGS= $(STD) $(WARN) $(OPT) $(DEBUG) $(CFLAGS) -fPIC
 R_LDFLAGS= $(LDFLAGS)
 DEBUG= -g
 
diff -Naurp a/deps/linenoise/Makefile b/deps/linenoise/Makefile
--- a/deps/linenoise/Makefile	2016-05-06 02:11:36.000000000 -0500
+++ b/deps/linenoise/Makefile	2016-06-01 18:06:58.691789064 -0500
@@ -2,7 +2,7 @@ STD=
 WARN= -Wall
 OPT= -Os
 
-R_CFLAGS= $(STD) $(WARN) $(OPT) $(DEBUG) $(CFLAGS)
+R_CFLAGS= $(STD) $(WARN) $(OPT) $(DEBUG) $(CFLAGS) -fPIC
 R_LDFLAGS= $(LDFLAGS)
 DEBUG= -g
 
diff -Naurp a/deps/Makefile b/deps/Makefile
--- a/deps/Makefile	2016-05-06 02:11:36.000000000 -0500
+++ b/deps/Makefile	2016-06-01 18:06:58.691789064 -0500
@@ -59,7 +59,7 @@ ifeq ($(uname_S),SunOS)
 	LUA_CFLAGS= -D__C99FEATURES__=1
 endif
 
-LUA_CFLAGS+= -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC='' $(CFLAGS)
+LUA_CFLAGS+= -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC='' $(CFLAGS) -fPIC
 LUA_LDFLAGS+= $(LDFLAGS)
 # lua's Makefile defines AR="ar rcu", which is unusual, and makes it more
 # challenging to cross-compile lua (and redis).  These defines make it easier
